### PR TITLE
DEI-392: fix GitHub action warnings

### PR DESCRIPTION
**Pull request by erikj540**
## JIRA ticket

https://tetrascience.atlassian.net/browse/DEI-392

## Description

- Updates to fix some of the warnings we are seeing from our GitHub actions
- `action/checkout@v2` and `action/setup-python@v2` use a deprecated version of NodeJS. So I updated those actions to later versions that do not use the deprecated version of NodeJS. Here's [an article](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/) that describes the issue.
- Also using `set-output` is being deprecated so I updated instances of `set-output` to the new method of managing output. Here's [an article](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) that explains the issue

## Supporting test data

- [This task script PR](https://github.com/tetrascience/ts-task-script-mettler-toledo-labx-raw-to-ids/pull/57) is using this branch's action `.yml` files. The warnings in the "Code Quality" and "Publish" actions appear to be fixed.
- [This IDS PR](https://github.com/tetrascience/ts-task-script-mettler-toledo-labx-raw-to-ids/pull/57) is using this branch's action `.yml` files. The warnings in the "IDS Quality Checks" and "Publish" actions appear to be fixed.
- I tested the `publish_tag.yml` and `publish_pypi_package.yml` changes on a test IDS repo. [Here are those actions](https://github.com/tetrascience/ts-ids-dei-392-test/actions/runs/5215334983) which crucially have no warning 

## Automated testing

- [ ] Unit tests
- [ ] Integration tests
- [ ] API tests
- [ ] End-To-End tests
- [ ] N/A (please clarify)

### CODE REVIEW

File: .github/workflows/publish_tag.yml [hunk 1]
```diff
@@ -411,7 +411,7 @@ jobs:
         working-directory: artifact_code
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python
```
Comment (mlipschultz) [inline .github/workflows/publish_tag.yml:414]: Should this be
```suggestion
        uses: actions/setup-python@v3
```

File: .github/workflows/ids_code_quality.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - name: "Set up Python"
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v3
```
Comment (mlipschultz) [inline .github/workflows/ids_code_quality.yml:16]: Any reason to you v3 over [the latest (v4)](https://github.com/actions/setup-python)?

**Review by mlipschultz — COMMENTED** [review]
I think there's one missing change and I have a question about the version for `setup-python`

File: .github/workflows/ids_code_quality.yml (hunk 2 above)
Comment (erikj540) [reply .github/workflows/ids_code_quality.yml]: Nope, no reason. I updated the version to v4.

**Review by nilsguillermin — APPROVED** [review]

**Pull request merged**